### PR TITLE
Simple Domain Resources Construct and a Construct that Can Add Behaviors to Those Resources

### PR DESCRIPTION
**Pull request by MarteagaJ**

### CODE REVIEW

**Comment by traviswalker26** [comment]
This is looking really good. My only question is would it be easier since this construct isn't being instantiated directly from frankenstack is would it be easier instead of listing all the domains as separate parameters to just make the domains parameter an array that you could iterate through?

**Comment by MarteagaJ** [comment]
> This is looking really good. My only question is would it be easier since this construct isn't being instantiated directly from frankenstack is would it be easier instead of listing all the domains as separate parameters to just make the domains parameter an array that you could iterate through?

@traviswalker26  Are we ok then with definitively deciding that the constructs within the open source repos will not be instantiated with Frank ever?  And the only frank instantiations will be done via the dsicollection specific construct?  In that case, then yes I can simplify the interfaces by making the domains passed in to the simple construct an array.  I can then make the CdkCloudfrontAddS3OriginProps and the CdkCloudfrontAddHttpOriginProps within the cdk-cloudfront-behavior construct take an array also.

File: packages/cdk-cloudfront-behavior/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  public distribution: cloudfront.Distribution;
+
+  constructor(scope: cdk.Construct, id: string, props: CdkCloudfrontBehaviorProps) {
+    super(scope, id);
+    let defaultBehaviorOptions: cloudfront.BehaviorOptions
+
+    if (props.defaultBehaviorOrigin == "s3" && props.defaultS3OriginBucketName && props.defaultS3OriginPath && props.defaultOriginAccessIdentity) {
+      const codeBucket = s3.Bucket.fromBucketName(this, "DefaultOriginBucket", props.defaultS3OriginBucketName);
+
+      const originAccessIdentity = cloudfront.OriginAccessIdentity.fromOriginAccessIdentityName(this, `${props.defaultS3OriginBucketName.split(".").join("-")}-DefaultOriginAcessIdentity`, props.defaultOriginAccessIdentity);
```
Comment (ethanwe) [inline packages/cdk-cloudfront-behavior/lib/index.ts:53]: typo on DefaultOriginAcessIdentity

File: packages/cdk-cloudfront-behavior/lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    this.distribution = baseResources.distribution;
+  }
+
+  addS3OriginBehavior(props: CdkCloudfrontAddS3OriginProps): void {
+    const s3Origins = props.origins;
+
+    s3Origins.forEach(origin => {
+      const codeBucket = s3.Bucket.fromBucketName(this, `${origin.bucketName.split(".").join("-")}-OriginBucket`, origin.bucketName);
+
+      const originAccessIdentity = cloudfront.OriginAccessIdentity.fromOriginAccessIdentityName(this, `${origin.bucketName.split(".").join("-")}-OriginAcessIdentity`, origin.originAccessIdentity);
```
Comment (ethanwe) [inline packages/cdk-cloudfront-behavior/lib/index.ts:128]: typo on OriginAcessIdentity

File: packages/cdk-cloudfront-behavior/lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        origin: new origins.S3Origin(codeBucket, {
+          originPath: props.defaultS3OriginPath,
+          originAccessIdentity: originAccessIdentity
+        }),
+        allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL,
+        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+        cachePolicy: cachePolicy
+      }
+    }
+    // else if(props.defaultBehaviorOrigin == "load-balancer"){
```
Comment (ethanwe) [inline packages/cdk-cloudfront-behavior/lib/index.ts:75]: should we delete these comments?

**Review by traviswalker26 — APPROVED** [review]

**Review by traviswalker26 — APPROVED** [review]

**Pull request merged**